src/mongocxx/pool.cpp

```diff
@@ -132,5 +132,16 @@ stdx::optional<pool::entry> pool::try_acquire() {
         entry::unique_client(new client(cli), [this](client* client) { _release(client); }));
 }
 
+void* pool::construct_client_pool(const uri& uri) {
```

Comment:
`const` members of a class must be set in the initializer list of the constructor. Because we want to be able to call, check, and possibly throw an exception before setting `_impl`, this must be done in a separate function to ensure that we can still set `_impl` in the initializer list upon successful creation of a client pool.